src/thread.c

```diff
@@ -55,17 +55,21 @@ void MTY_RWLockDestroy(MTY_RWLock **rwlock)
 		return;
 
 	MTY_RWLock *ctx = *rwlock;
+	*rwlock = NULL;
+	MTY_MemoryBarrier();
```

Comment:
Just as a wild thought, it would be neat to have an `MTY_SecureNull(...)` function, which would have a this definition:
```c
void *MTY_SecureNull(void **ptr)
{
	void *old_ptr = *ptr;
	*ptr = NULL;
	MTY_MemoryBarrier();
	return old_ptr;
}
```
And taking that block as an example, you'd simply do instead: `MTY_RWLock *ctx = MTY_SecureNull(rwlock);`

_Disclaimer: I don't know how memory barriers work internally, i.e. if they're scoped to the executing function or not. If they are, my suggestion just make no sense, please ignore._

Reply:
What you're actually after here is an atomic pointer swap, which Bryan does have code for. But I just want to get this work in, because it covers the main issue. Atomically swapping the pointer is less of an improvement (but still one)

See https://github.com/parsec-cloud/libmatoya/commits/thread-safety-barrier/